airgun/views/common.py

```diff
@@ -445,6 +446,39 @@ def search(self, query):
         return self.table.read()
 
 
+class SearchableViewMixinPF4(SearchableViewMixin):
+    """Mixin which adds :class:`airgun.widgets.Search` widget and
+    :meth:`airgun.widgets.Search.search` to your view. It's useful for _most_ entities list views
+    where searchbox and results table are present.
+    Note that class which uses this mixin should have :attr: `table` attribute.
+    """
+
+    searchbox = PF4Search()
+    welcome_message = Text("//div[@class='blank-slate-pf' or @id='welcome']")
+
+    def search(self, query):
+        """Perform search using searchbox on the page and return table
+        contents.
+
+        :param str query: search query to type into search field. E.g. ``foo``
+            or ``name = "bar"``.
+        :return: list of dicts representing table rows
+        :rtype: list
+        """
+        if not hasattr(self.__class__, 'table'):
+            raise AttributeError(
+                f'Class {self.__class__.__name__} does not have attribute "table". '
+                'SearchableViewMixin only works with views, which have table for results. '
+                'Please define table or use custom search implementation instead'
+            )
+        if not self.is_searchable():
+            return None
+        self.searchbox.search(query)
+        # TODO: may need a sleeper or a wait_for
+        time.sleep(3)
```

Comment:
wait_for is always better if there's something on the table you can use.

`self.table.wait_displayed` maybe?